src/elevenlabs/conversational_ai/conversation.py

```diff
@@ -52,22 +52,44 @@ def interrupt(self):
         """
         pass
 
+
+class ClientTools:
+    def __init__(self):
+        self.tools = {}
+        self.lock = threading.Lock()
+
+    def register(self, tool_name, handler):
```

Comment:
Can you add type annotation here please?

Also given the call blocks the processing of audio maybe add a comment saying the tool call should finish quickly. I think eventually we should turn this interface async and support async tools, then we can return the response when ready.

Reply:
Good shout! added.